File: apps/admin/src/types/launch.ts
import type { LL2LaunchNormal } from './ll2-launch';

export interface Launch {
  id: string;
  external_id: string;
  background_image: string;
  image_list: string[];
  thumb_image: string;
}

export interface LaunchSerializer extends Launch {
  data: LL2LaunchNormal;
}

export interface LaunchList {
  count: number;
  launches: LaunchSerializer[];
}

File: services/backend/internal/models/launch.go
package models

// for DB
type Launch struct {
	ID              int64    `bson:"id" json:"id,string"`
	ExternalID      string   `bson:"external_id" json:"external_id"` // ll2 launch id
	BackgroundImage string   `bson:"background_image" json:"background_image"`
	ImageList       []string `bson:"image_list" json:"image_list"`
	ThumbImage      string   `bson:"thumb_image" json:"thumb_image"`
}

type LaunchSerializer struct {
	Launch
	Data LL2LaunchNormal `json:"data"`
}

type LaunchList struct {
	Count    int                `json:"count"`
	Launches []LaunchSerializer `json:"launches"`
}
